bot/check-result.sh

```diff
@@ -0,0 +1,483 @@
+#!/bin/bash
+#
+# Script to check the result of building the EESSI software layer.
+# Intended use is that it is called by a (batch) job running on a compute
+# node.
+#
+# This script is part of the EESSI software layer, see
+# https://github.com/EESSI/software-layer.git
+#
+# author: Thomas Roeblitz (@trz42)
+#
+# license: GPLv2
+#
+
+# result cases
+
+#  - SUCCESS (all of)
+#    - working directory contains slurm-JOBID.out file
+#    - working directory contains eessi*tar.gz
+#    - no message ERROR
+#    - no message FAILED
+#    - no message ' required modules missing:'
+#    - one or more of 'No missing modules!'
+#    - message regarding created tarball
+#  - FAILED (one of ... implemented as NOT SUCCESS)
+#    - no slurm-JOBID.out file
+#    - no tarball
+#    - message with ERROR
+#    - message with FAILED
+#    - message with ' required modules missing:'
+#    - no message regarding created tarball
+
+# stop as soon as something fails
+# set -e
+
+TOPDIR=$(dirname $(realpath $0))
+
+source ${TOPDIR}/../scripts/utils.sh
+source ${TOPDIR}/../scripts/cfg_files.sh
+
+# defaults
+export JOB_CFG_FILE="${JOB_CFG_FILE_OVERRIDE:=./cfg/job.cfg}"
+
+# check if ${JOB_CFG_FILE} exists
+if [[ ! -r "${JOB_CFG_FILE}" ]]; then
+    echo_red "job config file (JOB_CFG_FILE=${JOB_CFG_FILE}) does not exist or not readable"
+else
+    echo "bot/check-result.sh: showing ${JOB_CFG_FILE} from software-layer side"
+    cat ${JOB_CFG_FILE}
+
+    echo "bot/check-result.sh: obtaining configuration settings from '${JOB_CFG_FILE}'"
+    cfg_load ${JOB_CFG_FILE}
+fi
+
+display_help() {
+  echo "usage: $0 [OPTIONS]"
+  echo " OPTIONS:"
+  echo "  -h | --help    - display this usage information [default: false]"
+  echo "  -v | --verbose - display more information [default: false]"
+}
+
+# set defaults for command line arguments
+VERBOSE=0
+
+POSITIONAL_ARGS=()
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -h|--help)
+      display_help
+      exit 0
+      ;;
+    -v|--verbose)
+      VERBOSE=1
+      shift 1
+      ;;
+    --)
+      shift
+      POSITIONAL_ARGS+=("$@") # save positional args
+      break
+      ;;
+    -*|--*)
+      fatal_error "Unknown option: $1" "${CMDLINE_ARG_UNKNOWN_EXITCODE}"
+      ;;
+    *)  # No more options
+      POSITIONAL_ARGS+=("$1") # save positional arg
+      shift
+      ;;
+  esac
+done
+
+set -- "${POSITIONAL_ARGS[@]}"
+
+job_dir=${PWD}
+
+[[ ${VERBOSE} -ne 0 ]] && echo ">> analysing job in directory ${job_dir}"
+
+GP_slurm_out="slurm-${SLURM_JOB_ID}.out"
+job_out=$(ls ${job_dir} | grep "${GP_slurm_out}")
```

Comment:
Why use `ls` at all, why not simply check with `-f` ?
Note that with the `ls` + `grep` approach, you should have a partial match, in theory (`foo_slurm-1234.out.txt` would also result in `SLURM=1`, while it's not the expected file) 

```shell
GP_slurm_out="slurm-${SLURM_JOB_ID}.out"
[ ${VERBOSE} -ne 0 ]] && echo ">> searching for job output file(s) matching '"${GP_slurm_out}"'"
if [[ -f ${GP_slurm_out} ]]; then
    SLURM=1
    [[ ${VERBOSE} -ne 0 ]] && echo "   found Slurm output file '"${job_out}"'"
else
    SLURM=0
    [[ ${VERBOSE} -ne 0 ]] && echo "   Slurm output file '"${job_out}"' NOT found!"
fi
```

With the current approach, you'll always print  a "`found slurm output file`" message, even when the Slurm output file was not found, which is a bit misleading...

Reply:
Using check with `-f` now. The misleading message was a bug. Should be fixed now.